presto-hive/src/test/java/io/prestosql/plugin/hive/s3/TestS3SecurityMapping.java

```diff
@@ -195,6 +196,69 @@ public void testMapping()
                 role("danny_hq_role"));
     }
 
+    @Test
+    public void testMappingWithFallbackToClusterDefault()
+    {
+        S3SecurityMappingConfig mappingConfig = new S3SecurityMappingConfig()
+                .setConfigFile(new File(getResource(getClass(), "security-mapping-with-fallback-to-cluster-default.json").getPath()));
+
+        DynamicConfigurationProvider provider = new S3SecurityMappingConfigurationProvider(mappingConfig);
+
+        // matches prefix - returns role from the mapping
+        assertMapping(
+                provider,
+                path("s3://bar/abc/data/test.csv"),
+                role("arn:aws:iam::123456789101:role/allow_path"));
+
+        // doesn't match any rule except default rule at the end
+        assertMapping(
+                provider,
+                empty(),
+                clusterDefaultRole());
+    }
+
+    @Test
+    public void testMappingWithoutFallback()
+    {
+        S3SecurityMappingConfig mappingConfig = new S3SecurityMappingConfig()
+                .setConfigFile(new File(getResource(getClass(), "security-mapping-without-fallback.json").getPath()));
+
+        DynamicConfigurationProvider provider = new S3SecurityMappingConfigurationProvider(mappingConfig);
+
+        // matches prefix - returns role from the mapping
+        assertMapping(
+                provider,
+                path("s3://bar/abc/data/test.csv"),
+                role("arn:aws:iam::123456789101:role/allow_path"));
+
+        // doesn't match any rule
+        assertMappingFails(
+                provider,
+                empty(),
+                "No matching S3 security mapping");
+    }
+
+    @Test
+    public void mappingWithoutRoleCredentialsFallbackShouldFail()
```

Comment:
```suggestion
    public void testMappingWithoutRoleCredentialsFallbackShouldFail()
```

Reply:
Do I have to stick to convention that each test method annotated with @Test should have prefix 'test'?

Reply:
Yes, please

Reply:
done